web/src/stores/clips.ts

```diff
@@ -53,93 +60,105 @@ export namespace Clips {
 
   export type Action =
     | LoadAction
+    | LoadErrorAction
     | RefillCacheAction
     | RemoveClipAction
     | AchievementAction;
 
   export const actions = {
-    refillCache: () => async (
-      dispatch: Dispatch<RefillCacheAction | LoadAction>,
-      getState: () => StateTree
-    ) => {
-      const state = getState();
-      if (localeClips(state).clips.length > MIN_CACHE_SIZE) {
-        return;
-      }
+    refillCache:
+      () =>
+      async (
+        dispatch: Dispatch<RefillCacheAction | LoadAction | LoadErrorAction>,
+        getState: () => StateTree
+      ) => {
+        const state = getState();
+
+        // don't load if no contributable locale
+        if (!contributableLocales.includes(state.locale)) {
+          return;
+        }
 
-      try {
-        dispatch({ type: ActionType.LOAD });
-        const clips = await state.api.fetchRandomClips(
-          MIN_CACHE_SIZE - localeClips(state).clips.length
-        );
+        if (localeClips(state).clips.length > MIN_CACHE_SIZE) {
+          return;
+        }
 
-        dispatch({
-          type: ActionType.REFILL_CACHE,
-          clips: clips.map(clip => {
-            let sentence = clip.sentence;
-            try {
-              sentence.text = decodeURIComponent(sentence.text);
-            } catch (e) {
-              if (e.name !== 'URIError') {
-                throw e;
+        try {
+          dispatch({ type: ActionType.LOAD });
+
+          const clips = await state.api.fetchRandomClips(
+            MIN_CACHE_SIZE - localeClips(state).clips.length
+          );
+
+          dispatch({
+            type: ActionType.REFILL_CACHE,
+            clips: clips.map(clip => {
+              const sentence = clip.sentence;
+              try {
+                sentence.text = decodeURIComponent(sentence.text);
+              } catch (e) {
+                if (e.name !== 'URIError') {
+                  throw e;
+                }
               }
-            }
-
-            return {
-              id: clip.id,
-              glob: clip.glob,
-              sentence,
-              audioSrc: clip.audioSrc,
-            };
-          }),
-        });
-        await Promise.all(clips.map(({ audioSrc }) => fetch(audioSrc)));
-      } catch (err) {
-        if (err instanceof XMLHttpRequest) {
-          dispatch({ type: ActionType.REFILL_CACHE });
-        } else {
-          throw err;
+
+              return {
+                id: clip.id,
+                glob: clip.glob,
+                sentence,
+                audioSrc: clip.audioSrc,
+              };
+            }),
+          });
+          await Promise.all(clips.map(({ audioSrc }) => fetch(audioSrc)));
```

Comment:
I think this is going to expose an issue that wasn't introduced by this PR but that will be made more obvious by it. The clips were originally loaded immediately on loading any page because `/api/v1/clips` doesn't actually return the clip files, just the public URLs, and the clips themselves aren't loaded until the URLs are returned to the store and they're fetched one by one. So clips were being loaded in immediately so that all of this downloading could happen in the background, so that they would be ready when users hit `/listen`. If clips weren't ready when users hit 'play', there would be all sorts of janky behaviour like the animation going wildly, unsupportedSource errors, etc. 

That was true before and is still true now, but I think moving the load to `/listen` (which makes logical sense) will expose that behaviour to more users, and the presence of a loader will make users think that the clips _have_ loaded when they haven't and be even more confused.  Based on local testing, it seems like the `refillCache` will return even if the fetch operations themselves error or time out. I don't think solving for that should block this from deploying, esp in terms of getting the site back to operations ASAP, but I wanted to flag it as something that will need a better solution soon. (Esp as AWS is also experiencing outages, which will mess with our clip loading.)

Reply:
This is really useful thank you, I added two commits which I think could help. Could you review those for me?

Reply:
This could make the page a bit slower as we have to wait for all of them to be loaded enough to be able to be played. So really a better solution would be to do this error checking for each clip, but I think this is better than what we currently have?

Reply:
I think that's a really good interim solution though, thanks for putting this together!